lib/data_update_scripts/20211115154021_nullify_invalid_tag_fields.rb

```diff
@@ -0,0 +1,34 @@
+module DataUpdateScripts
+  class NullifyInvalidTagFields
+    def run
+      Tag.where(bg_color_hex: "").or(Tag.where(text_color_hex: "")).find_each do |tag|
+        # presence here selects non-empty strings if set  (avoiding unsetting valid values)
+        # but nullifies empty tag colors
+        tag.update_columns(
+          bg_color_hex: tag.bg_color_hex.presence,
+          text_color_hex: tag.text_color_hex.presence,
+        )
+      end
```

Comment:
Is there a reason not to let the DB run with these the way you did below for `alias_for`?

```suggestion
      Tag.where(bg_color_hex: "").update(bg_color_hex: nil)
      Tag.where(text_color_hex: "").update(text_color_hex: nil)
```

Reply:
yes - in fact that was the first attempt last week - it was inneffective. 

the issue was we can't update `bg_color_hex` if `text_color_hex` is _also_ invalid/empty (since update calls validations), and all of the invalid records (in DEV's db) have both invalid.

I could split it out into `where(one column: '').update_column(:one_column, nil)` (separate queries, bypass validations) or use find_each tag, mutate the tag instance's attributes, and then try to save (triggering validations, possible to fail for other reasons I guess).

but can't do them in separate updates like that (since either value being invalid would prevent the other from saving).